sdk/modelsrepository/Azure.Iot.ModelsRepository/src/ResolverClient.cs

```diff
@@ -0,0 +1,129 @@
+
+namespace Azure.Iot.ModelsRepository
+{
+    /// <summary>
+    /// The <c>ResolverClient</c> class supports DTDL model resolution providing functionality to
+    /// resolve models by retrieving model definitions and their dependencies.
+    /// </summary>
+    public class ResolverClient
+    {
+        internal const string DefaultRepository = "https://devicemodels.azure.com";
+        private readonly RepositoryHandler _repositoryHandler;
+
+        /// <summary>
+        /// Initializes the <c>ResolverClient</c> with default client options while pointing to
+        /// the Azure IoT Plug and Play Model repository https://devicemodels.azure.com for resolution.
+        /// </summary>
+        public ResolverClient() : this(new Uri(DefaultRepository), null) { }
+
+        /// <summary>
+        /// Initializes the <c>ResolverClient</c> with default client options while pointing to
+        /// a custom <paramref name="repositoryUri"/> for resolution.
+        /// </summary>
+        /// <param name="repositoryUri">
+        /// The model repository <c>Uri</c> value. This can be a remote endpoint or local directory.
+        /// </param>
+        public ResolverClient(Uri repositoryUri) : this(repositoryUri, null) { }
+
+        /// <summary>
+        /// Initializes the <c>ResolverClient</c> with custom client <paramref name="options"/> while pointing to
+        /// the Azure IoT Plug and Play Model repository https://devicemodels.azure.com for resolution.
+        /// </summary>
+        /// <param name="options">
+        /// <c>ResolverClientOptions</c> to configure resolution and client behavior.
+        /// </param>
+        public ResolverClient(ResolverClientOptions options) : this(new Uri(DefaultRepository), options) { }
+
+        /// <summary>
+        /// Initializes the <c>ResolverClient</c> with custom client <paramref name="options"/> while pointing to
+        /// a custom <paramref name="repositoryUri"/> for resolution.
+        /// </summary>
+        /// <param name="repositoryUri">
+        /// The model repository <c>Uri</c>. This can be a remote endpoint or local directory.
+        /// </param>
+        /// <param name="options">
+        /// <c>ResolverClientOptions</c> to configure resolution and client behavior.
+        /// </param>
+        public ResolverClient(Uri repositoryUri, ResolverClientOptions options)
+        {
+            _repositoryHandler = new RepositoryHandler(repositoryUri, options);
+        }
+
+        /// <summary>
+        /// Initializes the <c>ResolverClient</c> with default client options while pointing to
+        /// a custom <paramref name="repositoryUriStr"/> for resolution.
+        /// </summary>
+        /// <param name="repositoryUriStr">
+        /// The model repository <c>Uri</c> in string format. This can be a remote endpoint or local directory.
+        /// </param>
+        public ResolverClient(string repositoryUriStr) : this(repositoryUriStr, null) { }
```

Comment:
We should deprecate this overload. We should take Uri type only

Reply:
task added